### ISSUE TITLE
multi-line string assignments reporting column errors

### ISSUE THREAD

**Issue by cxd4** [douglascrockford/JSLint#185]
```javascript
var GLSL_script =
    "attribute highp vec4 pos;\n" +
    "void main(void) {\n" +
    "    gl_Position = pos;\n" +
    "}\n";
```

JSLint complains:

* Expected 'attribute highp vec4 pos;\n' at column 8, not column 4.
* Expected 'void main(void) {\n' at column 8, not column 4.
* Expected ' gl_Position = pos;\n' at column 8, not column 4.
* Expected '}\n' at column 8, not column 4.

Should indents be 8 spaces or 4 spaces?

Non-ASCII multi-line assignments seem to say 4 spaces; ASCII ones say 8.  Was this intended?

**Comment by douglascrockford** [comment]
8

**Comment by cxd4** [comment]
OK, thanks.

Then can I ask you if this is meant to be consistent with two-dimensional arrays?
Here:
```javascript
var GLSL_scripts = [
    [ // 4-dimensional positions with constant color
                "attribute highp vec4 pos;\n" +
                "void main(void) {\n" +
                "    gl_Position = vec4(pos);\n" +
                "}\n",

                "uniform lowp vec4 const_color;\n" +
                "void main(void) {\n" +
                "    gl_FragColor = const_color;\n" +
                "}\n"
    ]
];
```
JSLint complains:

* Expected 'attribute highp vec4 pos;\n' at column 8, not column 16.
* Expected 'uniform lowp vec4 const_color;\n' at column 8, not column 16.

According to those messages, I can only fix the warnings by changing to this:
```javascript
var GLSL_scripts = [
    [ // 4-dimensional positions with constant color
        "attribute highp vec4 pos;\n" +
                "void main(void) {\n" +
                "    gl_Position = vec4(pos);\n" +
                "}\n",

        "uniform lowp vec4 const_color;\n" +
                "void main(void) {\n" +
                "    gl_FragColor = const_color;\n" +
                "}\n"
    ]
];
```

Just to make sure I'm clear:  Is this, too, what you have also intended?

**Comment by douglascrockford** [comment]
I recommend that you follow JSLint's advice.

**Comment by cxd4** [comment]
I imagine so, but that's the reason to ask if a bug is at work.

Such advice is meant to be followed when the actual suggestion is the intended suggestion.  I've followed all of JSLint's advice so far, but if a bug causes not 4- but 50-char indents, would we still recommend it?

Does anybody know of a post, article or documentation of some kind which explains these multi-line string indentation rules?  I am hoping there are good reasons relevant to the nature of JavaScript as a language that outweigh the loss of GLSL readability to my programs from adjusting the multi-line indentations in this fashion, because this has a jumpy and inconsistent appearance.

**Comment by cxd4** [comment]
update, just for perspective:  https://github.com/cxd4/OpenGL-js/commit/d6287cc5d7dae71b213f8b93fec4e062aba97232

Again, if people ask me why I made the spacing inconsistent, I'd like to have an explanation ready for them other than just, "Because JSLint said so."

**Comment by cxd4** [comment]
Last bump sorry, just in case it would help anyone else.

I've discovered and committed the solution of [ES6 multi-line strings](https://github.com/cxd4/OpenGL-js/commit/5ea765838f98450a16eaba37d1818ac6d86a9fb0) from JSLint's native toleration.  While I'm not all that happy about the diff in a few aspects, I find it to be overall an improvement over the warped indentations in Crockford's personal preference in the pre-ES6 way.

Actually needed the `/*jslint es6:  true*/` hack regardless due to WebGL's inevitable ES6 requirements for VRAM uploads to server-side vertex arrays, so, luckily, we're kind of killing 2 birds with 1 stone.
